MIP/mip-61/README.md

```diff
@@ -0,0 +1,185 @@
+The relayer can go offline or crash. It needs to automatically understand from where it should start with sending messages to the target chains, i.e. what is the lowest nonce for which the transfer is not complete.
+
+Currently, when the relayer starts, it doesn't reload the state that it has when it shuts down. If transfers are processed when the relayer stops, the transfers are lost and the relayer is not able to continue the transfer.
+
+Moreover, even if the data would be stored on where to continue, the relayer may crash for a variety of reasons. It needs to automatically understand from where it should start with sending messages to the target chains.
+
+## Context
+
+This algo is defined using the lock/mint-type Native Bridge protocol, described in [MIP-58](https://github.com/movementlabsxyz/MIP/pull/58).This bridge protocol has two function calls :
+
+- `Initiate_transfer` on the source chain (called by user)
+- `Complete_transfer` on the target chain (called by relayer)
+
+This algo is compatible with using finalized blocks (see the remark on L2 complete verification). By pulling only finalized source blocks, the  nonce are always updated on finalization.
+
+In this MIP we consider the lock/mint Native Bridge, see [MIP-60](https://github.com/movementlabsxyz/MIP/pull/60). We will discuss a single direction of transfer - from **source chain** to **target chain**.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+!!! . The algorithm is the same in both directions. Hence each algorithm has to be implemented twice. Once for the transfer direction L1 --> L2, and once for the L2 --> L1 direction.
+
+The relayer can get its state from the source chain and target chain. The algo has three parts:
+
+**1. Continuous-Processing**: The relayer pulls transfer events from the source chain and processes them. For each transfer the relayer has to check if the transfer is completed on the target chain, and if not it has to complete the transfer.
+
+**2. Bootstrap:** The relayer initiate the event polling at some point in the past. This point is either the genesis, some configurable source block height, or some checkpoint (possibly on-chain)
+
+#### Event-Processing
+
+The relayer manages only one state of transfer: pending, which exists during the time between the reception of the `initiate_transfer` event from the source chain and the success of the `complete_transfer` tx on the target chain.
+
+The Native Bridge protocol MUST implement the assignment on the source chain of an incrementing `nonce`. Since there are two directions there MUST be two counters - one for each direction. The `nonce` is used to order the transfers.
+
+For each `initiate_transfer` event do the following:
+
+![alt text](event_processing.png)
+
+1. Get the `nonce`.
```

Comment:
I think this would be better as or preceded by a pseudocode block. In general, I prefer either pseudo-<insert language we already use> or a university pseudocode convention.

Reply:
i agree. this has been created under time pressure. it should be converted to pseudocode when time permits